scrapy/utils/signal.py

```diff
@@ -5,75 +5,31 @@
 
 
-def send_catch_log(signal=Any, sender=Anonymous, *arguments, **named):
+def send_catch_log(signal=None, sender=None, *arguments, **named):
     """Like pydispatcher.robust.sendRobust but it also logs errors and returns
     Failures instead of exceptions.
     """
-    dont_log = named.pop('dont_log', _IgnoredException)
-    spider = named.get('spider', None)
-    responses = []
-    for receiver in liveReceivers(getAllReceivers(sender, signal)):
-        try:
-            response = robustApply(receiver, signal=signal, sender=sender,
-                *arguments, **named)
-            if isinstance(response, Deferred):
-                logger.error("Cannot return deferreds from signal handler: %(receiver)s",
-                             {'receiver': receiver}, extra={'spider': spider})
-        except dont_log:
-            result = Failure()
-        except Exception:
-            result = Failure()
-            logger.error("Error caught on signal handler: %(receiver)s",
-                         {'receiver': receiver},
-                         exc_info=True, extra={'spider': spider})
-        else:
-            result = response
-        responses.append((receiver, result))
-    return responses
+    return signal.send_robust(sender=sender, **named)
 
 
-def send_catch_log_deferred(signal=Any, sender=Anonymous, *arguments, **named):
+def send_catch_log_deferred(signal=None, sender=None, *arguments, **named):
     """Like send_catch_log but supports returning deferreds on signal handlers.
     Returns a deferred that gets fired once all signal handlers deferreds were
     fired.
     """
-    def logerror(failure, recv):
-        if dont_log is None or not isinstance(failure.value, dont_log):
-            logger.error("Error caught on signal handler: %(receiver)s",
-                         {'receiver': recv},
-                         exc_info=failure_to_exc_info(failure),
-                         extra={'spider': spider})
-        return failure
-
-    dont_log = named.pop('dont_log', None)
-    spider = named.get('spider', None)
-    dfds = []
-    for receiver in liveReceivers(getAllReceivers(sender, signal)):
-        d = maybeDeferred(robustApply, receiver, signal=signal, sender=sender,
-                *arguments, **named)
-        d.addErrback(logerror, receiver)
-        d.addBoth(lambda result: (receiver, result))
-        dfds.append(d)
-    d = DeferredList(dfds)
-    d.addCallback(lambda out: [x[1] for x in out])
-    return d
+    return signal.send_robust_deferred(sender=sender, **named)
 
 
-def disconnect_all(signal=Any, sender=Any):
+def disconnect_all(signal, sender=None):
     """Disconnect all signal handlers. Useful for cleaning up after running
     tests
     """
-    for receiver in liveReceivers(getAllReceivers(sender, signal)):
-        disconnect(receiver, signal=signal, sender=sender)
+    for receiver in signal._live_receivers(sender=sender):
+        signal.disconnect(receiver=receiver, sender=sender)
```

Comment:
This should be a method of `Signal`, with only a pass-through here (like the two functions above)